homeassistant/components/smartthings/config_flow.py

```diff
@@ -0,0 +1,186 @@
+"""Config flow to configure SmartThings."""
+import logging
+
+from aiohttp.client_exceptions import ClientResponseError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_ACCESS_TOKEN
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import (
+    CONF_APP_ID, CONF_INSTALLED_APP_ID, CONF_LOCATION_ID, DOMAIN,
+    VAL_UID_MATCHER)
+from .smartapp import (
+    create_app, find_app, setup_smartapp, setup_smartapp_endpoint, update_app)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class SmartThingsFlowHandler(config_entries.ConfigFlow):
+    """
+    Handle configuration of SmartThings integrations.
+
+    Any number of integrations are supported.  The high level flow follows:
+    1) Flow initiated
+        a) User initiates through the UI
+        b) Re-configuration of a failed entry setup
+    2) Enter access token
+        a) Check not already setup
+        b) Validate format
+        c) Setup SmartApp
+    3) Wait for Installation
+        a) Check user installed into one or more locations
+        b) Config entries setup for all installations
+    """
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_PUSH
+
+    def __init__(self):
+        """Create a new instance of the flow handler."""
+        self.access_token = None
+        self.app_id = None
+        self.api = None
+
+    async def async_step_import(self, user_input=None):
+        """Occurs when a previously entry setup fails and is re-initiated."""
+        return await self.async_step_user(user_input)
+
+    async def async_step_user(self, user_input=None):
+        """Get access token and validate it."""
+        from pysmartthings import SmartThings
+
+        errors = {}
+        if not self.hass.config.api.base_url.lower().startswith('https://'):
```

Comment:
This is more like an FYI, but since you use webhooks, it could also be a cloud webhook, and then the base url doesn't need this.

Reply:
Does that apply to the usage of webhooks in this component?  I have to know the full URL during the config flow as it's registered (and validated) by SmartThings during the creation of the app.